pkg/operation/botanist/component/kubestatemetrics/kube_state_metrics_test.go

```diff
@@ -61,6 +62,68 @@ var _ = Describe("KubeStateMetrics", func() {
 
 		serviceAccount    *corev1.ServiceAccount
 		secretShootAccess *corev1.Secret
+		clusterRoleFor    = func(clusterType component.ClusterType) *rbacv1.ClusterRole {
+			name := "gardener.cloud:monitoring:kube-state-metrics"
+			if clusterType == component.ClusterTypeSeed {
+				name += "-seed"
+			}
+
+			obj := &rbacv1.ClusterRole{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "rbac.authorization.k8s.io/v1",
+					Kind:       "ClusterRole",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+					Labels: map[string]string{
+						"component": "kube-state-metrics",
+						"type":      string(clusterType),
+					},
+				},
+				Rules: []rbacv1.PolicyRule{
+					{
+						APIGroups: []string{""},
+						Resources: []string{
+							"nodes",
+							"pods",
+							"services",
+							"resourcequotas",
+							"replicationcontrollers",
+							"limitranges",
+							"persistentvolumeclaims",
+							"namespaces",
+						},
+						Verbs: []string{"list", "watch"},
+					},
+					{
+						APIGroups: []string{"apps", "extensions"},
+						Resources: []string{"daemonsets", "deployments", "replicasets", "statefulsets"},
+						Verbs:     []string{"list", "watch"},
+					},
+					{
+						APIGroups: []string{"batch"},
+						Resources: []string{"cronjobs", "jobs"},
+						Verbs:     []string{"list", "watch"},
+					},
+				},
+			}
+
+			if clusterType == component.ClusterTypeSeed {
```

Comment:
Can we rather have the 2 expected roles as vars instead of duplicating the original logic in the test?

Reply:
Why/for which benefit? In the end, we only care about the fact that the productive code generates the proper `ClusterRole` object, no matter how it's generated in the test?